src/clusterfuzz/_internal/base/concurrency.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tools for concurrency/parallelism."""
+from concurrent import futures
+import contextlib
+import multiprocessing
+
+from clusterfuzz._internal.system import environment
+
+POOL_SIZE = multiprocessing.cpu_count()
+
+
+@contextlib.contextmanager
+def make_pool(pool_size=POOL_SIZE):
+  # Don't use processes on Windows and unittests to avoid hangs.
```

Comment:
Do you know why we hang? is this because of the start method for multiprocessing being "spawn" on windows? or something else?

Reply:
I'm not sure to be honest. The comment isn't really detailed enough for me to know exactly what happened. My mistake.